### PR TITLE
Split out trigger and broker docs

### LINKED ISSUE #2554: Split out broker and trigger docs
**Describe the change you'd like to see**

- [x] Break out broker and trigger docs into their own sections
- [x] Update broker docs to remove default broker, this is now MT broker
- [x] General cleanup of broker and trigger user stories

**Additional context**
Follow up comments from #2349 
cc @vaikas

### PR DESCRIPTION
**Pull request by abrennan89**
Partially fixes # 2554 
## Proposed Changes

- Split broker and trigger docs into their own sections at top level.

### CODE REVIEW

File: docs/eventing/triggers/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: my-service
+EOF
+```
+
+The example above filters events from the `default` Broker that are of type
+`dev.knative.foo.bar` AND have the extension `myextension` with the value
+`my-extension-value`.
```
Comment (mattmoor) [inline docs/eventing/triggers/README.md:53]: Remove trailing whitespace:
```suggestion
`my-extension-value`.
```

Comment (mattmoor) [inline docs/eventing/triggers/README.md:53]: Remove trailing whitespace:
```suggestion
`my-extension-value`.
```

Comment (mattmoor) [inline docs/eventing/triggers/README.md:53]: Remove trailing whitespace:
```suggestion
`my-extension-value`.
```

Comment (mattmoor) [inline docs/eventing/triggers/README.md:53]: Remove trailing whitespace:
```suggestion
`my-extension-value`.
```

**Comment by vaikas** [comment]
/approve
Thanks!!

**Pull request merged**